softlayer/data_source_softlayer_vlan.go

```diff
@@ -0,0 +1,74 @@
+package softlayer
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+)
+
+func dataSourceSoftLayerVlan() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSoftLayerVlanRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"name": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"number", "router_hostname"},
+			},
+
+			"number": {
+				Type:          schema.TypeInt,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
+
+			"router_hostname": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"name"},
+			},
```

Comment:
I think that consistent schema names should be used for data and resources. Field names `number` and `router_houstname` are different from the field names in a vlan resource.